src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const arr = this;
+
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] !== undefined && arr[i] !== null) {
+        result += arr[i];
+      } else {
+        result += '';
+      }
```

Comment:
it's a redundant else statement
```suggestion
      } 
```